.github/workflows/publish.yml

```diff
@@ -0,0 +1,66 @@
+name: Publish artifacts
+
+on:
+  release:
+    ### NOTE:
+    ###
+    ### There are multiple types of release trigger events.
+    ### Ref: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release
+    ###
+    ### Given that GitHub allows the creation of 'release'
+    ### and 'pre-release' publications to differentiate between
+    ### production and beta releases, the trigger type to use,
+    ### as it wraps them both, must be 'published'.
+    types: [published]
+
+jobs:
+
+  docker:
+    needs: []
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Set up GitHub Actions"
+      uses: actions/checkout@v2
+    - name: "Login to DockerHub"
+      uses: docker/login-action@v1
+      with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: "Build and publish to registry"
+      uses: docker/build-push-action@v2
+      with:
+        push: true
+        context: .
+        tags: madminertool/docker-madminer:latest
+        labels: |
+          org.opencontainers.image.source=${{ github.event.repository.html_url }}
+          org.opencontainers.image.created=${{ steps.prep.outputs.created }}
+          org.opencontainers.image.revision=${{ github.sha }}
+
+  package:
+    needs: []
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Set up GitHub Actions"
+      uses: actions/checkout@v2
+    - name: "Set up Python 3.6"
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    - name: "Install Python dependencies"
+      run: |
+        pip install --no-cache-dir --upgrade pip setuptools wheel
+        pip install --no-cache-dir --upgrade twine
+        pip install --no-cache-dir --quiet .
+        pip list
+    - name: "Build Python package"
+      run: |
+        python setup.py sdist bdist_wheel --universal
```

Comment:
Can you use [`build`](https://pypi.org/project/build/) instead of needing to call `setup.py` directly? c.f. https://github.com/scikit-hep/pyhf/blob/70f100c56588e4119eca5d00ddb101911a4cd50e/.github/workflows/publish-package.yml#L25-L34

Also you're using the `--universal` flag, but `madminer` is Python 3 only now. c.f. https://github.com/scikit-hep/pyhf/pull/1295

Reply:
Package publication command replaced in commit beef0b105a9fb7a78a90244332c87442ede6ccfc

Reply:
Yeah, it would be exactly that. Though `build` uses PEP 517/518 so you'd also need to [specify your build system for it](https://github.com/scikit-hep/pyhf/blob/70f100c56588e4119eca5d00ddb101911a4cd50e/pyproject.toml#L1-L4).

So for you, your `pyproject.toml` would probably only need to be

```toml
[build-system]
# Minimum requirements for the build system to execute.
requires = ["wheel", "setuptools>=30.3.0"]
build-backend = "setuptools.build_meta"
```

I'd strongly suggest that you have these run as part of the CI on PRs but skip the deploy stage, and then only deploy on actual releases. c.f. https://github.com/scikit-hep/pyhf/blob/master/.github/workflows/publish-package.yml for an example of that. Far better to have your build attempt fail in CI where you can then diagnose it safely, then when you are in the process of deploying :grimacing:

Reply:
> Package publication command replaced in commit [beef0b1](https://github.com/diana-hep/madminer/commit/beef0b105a9fb7a78a90244332c87442ede6ccfc)

Nice. :) This will fail without the build-system specification described above though.